modules/src/ics07_tendermint/error.rs

```diff
@@ -1,5 +1,70 @@
 use crate::ics24_host::error::ValidationError;
 use flex_error::{define_error, DisplayOnly, TraceError};
+use std::fmt::Display;
+
+#[derive(Debug)]
+pub struct StdError {
+    inner: Box<dyn std::error::Error + Send + Sync>,
+}
+
+impl StdError {
+    pub fn new(value: Box<dyn std::error::Error + Send + Sync>) -> Self  {
+        Self {
+            inner: value
+        }
+    }
+
+}
+
+impl From<String> for StdError {
+    fn from(value: String) -> Self {
+        Self {
+            inner: value.into()
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum LegacyError {
+    InnerBox(StdError),
+    #[cfg(feature = "test_no_std")]
+    InnerString(String),
+}
+
+impl LegacyError {
+    fn as_str(&self) -> String {
```

Comment:
We can have a [`ToString`](https://doc.rust-lang.org/std/string/trait.ToString.html) implementation of `LegacyError`.